plugin-server/src/cdp/consumers/cdp-source-webhooks.consumer.ts

```diff
@@ -0,0 +1,154 @@
+import express from 'express'
+import { DateTime } from 'luxon'
+
+import { Hub } from '../../types'
+import { logger } from '../../utils/logger'
+import { PromiseScheduler } from '../../utils/promise-scheduler'
+import { UUIDT } from '../../utils/utils'
+import { buildGlobalsWithInputs } from '../services/hog-executor.service'
+import { CyclotronJobQueue } from '../services/job-queue/job-queue'
+import {
+    CyclotronJobInvocationHogFunction,
+    CyclotronJobInvocationResult,
+    HogFunctionInvocationGlobals,
+    HogFunctionType,
+} from '../types'
+import { createInvocation, createInvocationResult } from '../utils/invocation-utils'
+import { CdpConsumerBase } from './cdp-base.consumer'
+
+const getFirstHeaderValue = (value: string | string[] | undefined): string | undefined => {
+    return Array.isArray(value) ? value[0] : value
+}
+
+export class CdpSourceWebhooksConsumer extends CdpConsumerBase {
+    protected name = 'CdpSourceWebhooksConsumer'
+    private cyclotronJobQueue: CyclotronJobQueue
+    private promiseScheduler: PromiseScheduler
+
+    constructor(hub: Hub) {
+        super(hub)
+        this.promiseScheduler = new PromiseScheduler()
+        this.cyclotronJobQueue = new CyclotronJobQueue(hub, 'hog')
+    }
+
+    public async getWebhook(webhookId: string): Promise<HogFunctionType | null> {
+        const hogFunction = await this.hogFunctionManager.getHogFunction(webhookId)
+
+        if (hogFunction?.type !== 'source_webhook') {
+            return null
+        }
+
+        return hogFunction
+    }
+
+    public async processWebhook(webhookId: string, req: express.Request) {
+        const hogFunction = await this.getWebhook(webhookId)
+
+        if (!hogFunction) {
+            // TODO: Maybe better error types?
+            throw new Error('Not found')
+        }
+
+        const headers: Record<string, string> = {}
+
+        for (const [key, value] of Object.entries(req.headers)) {
+            // TODO: WE should filter the headers to only include ones we know are safe to expose
+            const firstValue = getFirstHeaderValue(value)
+            if (firstValue) {
+                headers[key.toLowerCase()] = firstValue
+            }
+        }
+
+        const body: Record<string, any> = req.body
+        // TODO: Should this be filled via other headers?
```

Comment:
I have a lot of TODOs here that are purposefully to be fixed later. I'll have an issue up about remaining work for this to be releasable